src/lsdb/dask/merge_catalog_functions.py

```diff
@@ -71,7 +76,60 @@ def align_catalogs(left: Catalog, right: Catalog, add_right_margin: bool = True)
     Returns:
         The PixelAlignment object from aligning the catalogs
     """
+    right_tree, right_moc = _get_right_tree_and_moc(right, add_right_margin)
+    return align_with_mocs(
+        left.hc_structure.pixel_tree,
+        right_tree,
+        left.hc_structure.moc,
+        right_moc,
+        alignment_type=PixelAlignmentType.INNER,
+    )
+
 
+def align_catalogs_with_association(
```

Comment:
After discussing this morning, this approach makes even more sense, and I like it.